utils/scarb-ui/src/lib.rs

```diff
@@ -119,6 +119,24 @@ impl Ui {
         self.print(TypedMessage::styled("error", "red", message.as_ref()))
     }
 
+    /// Print a warning to the user.
+    pub fn warn_with_code(&self, code: impl AsRef<str>, message: impl AsRef<str>) {
+        self.print(TypedMessage::styled(
+            format!("warn[{}]", code.as_ref()).as_str(),
+            "yellow",
+            message.as_ref(),
+        ))
+    }
+
+    /// Print an error to the user.
+    pub fn error_with_code(&self, code: impl AsRef<str>, message: impl AsRef<str>) {
+        self.print(TypedMessage::styled(
+            format!("error[{}]", code.as_ref()).as_str(),
+            "red",
+            message.as_ref(),
+        ))
+    }
+
```

Comment:
This will print `error[CODE]` as message type in JSON output which is definitely not something expected